src/pages/levels/store.js

```diff
@@ -1,29 +1,259 @@
 /* eslint-disable no-param-reassign */
-import { action, thunk } from 'easy-peasy';
+import { action, thunk, actionOn } from 'easy-peasy';
 import {
   LevelPacks,
   LevelPackFavAdd,
   LevelPackFavRemove,
   LevelPackFavs,
   Collections,
+  LevelPacksStats,
 } from 'api';
+import { orderBy, partition, isEmpty } from 'lodash';
+import memoize from 'fast-memoize';
+import { shiftedLogisticFn } from 'utils/calcs';
+
+/**
+ *
+ * @param {Array} packs
+ * @param {Array} favs
+ * @param {Array} stats
+ * @param {string} sort
+ * @returns {*[]|*}
+ */
+const sortPacks = (packs, favs, stats, sort) => {
+  if (isEmpty(packs)) {
+    return [];
+  }
+
+  // deep clone
+  packs = JSON.parse(JSON.stringify(packs));
```

Comment:
`packs = {...packs};` should work just as well, and it's bad practice to reassign parameter variable

Reply:
in this case its an array of objects, and I think those objects might have some objects themselves. So a deep clone is necessary. I don't think the spread operator will do this although i haven't bothered to actually double check on that, but seems very unlikely to me.

Reply:
https://lodash.com/docs/4.17.15#cloneDeep is also an option.

Reply:
that would work fine too but i'm sure it's a micro optimization either way. In this case, packs is only data, it won't contain circular references.

Reply:
changed the name of function param to avoid re-assign.